In [3]:
import numpy as np
import pandas as pd
import io
import statsmodels.formula.api as smf
import scipy.stats as stats

In [4]:
data_string = '''Names        Estimate   Std.Error
(Intercept) 3446.830     81.840
totwrk        -0.169      0.018
age            2.714      1.472
south        102.272     41.925
male          87.108     35.173
smsa         -54.188     33.193
yngkid       -13.051     50.459
marr          31.360     42.263
union         11.866     38.186'''
data_string

'Names        Estimate   Std.Error\n(Intercept) 3446.830     81.840\ntotwrk        -0.169      0.018\nage            2.714      1.472\nsouth        102.272     41.925\nmale          87.108     35.173\nsmsa         -54.188     33.193\nyngkid       -13.051     50.459\nmarr          31.360     42.263\nunion         11.866     38.186'

In [5]:
df = pd.read_csv(io.StringIO(data_string), sep ='\s+')
df

Names  Estimate  Std.Error
0  (Intercept)  3446.830     81.840
1       totwrk    -0.169      0.018
2          age     2.714      1.472
3        south   102.272     41.925
4         male    87.108     35.173
5         smsa   -54.188     33.193
6       yngkid   -13.051     50.459
7         marr    31.360     42.263
8        union    11.866     38.186

In [55]:
df['t_nabl'] = np.abs(df['Estimate']/df['Std.Error'])
df

Names  Estimate  Std.Error     t_nabl
0  (Intercept)  3446.830     81.840  42.116691
1       totwrk    -0.169      0.018   9.388889
2          age     2.714      1.472   1.843750
3        south   102.272     41.925   2.439404
4         male    87.108     35.173   2.476559
5         smsa   -54.188     33.193   1.632513
6       yngkid   -13.051     50.459   0.258646
7         marr    31.360     42.263   0.742020
8        union    11.866     38.186   0.310742

In [7]:
alpha = 0.05
t_crit = stats.t.ppf(1-0.05/2, 707 - (8+1)).round(3)
t_crit

1.963

In [8]:
df['Znachimosty'] = df.apply(lambda x: 'znachim' if x['t_nabl']>t_crit else 'ne_znachim', axis = 1)
df

Names  Estimate  Std.Error     t_nabl Znachimosty
0  (Intercept)  3446.830     81.840  42.116691     znachim
1       totwrk    -0.169      0.018   9.388889     znachim
2          age     2.714      1.472   1.843750  ne_znachim
3        south   102.272     41.925   2.439404     znachim
4         male    87.108     35.173   2.476559     znachim
5         smsa   -54.188     33.193   1.632513  ne_znachim
6       yngkid   -13.051     50.459   0.258646  ne_znachim
7         marr    31.360     42.263   0.742020  ne_znachim
8        union    11.866     38.186   0.310742  ne_znachim

## Задача 1.2 (sleep equation, Python)

In [9]:
date_string = '''               Estimate  Std. Error t value  Pr(>|t|)    
(Intercept) 3431.803558   96.309432 35.6331 < 2.2e-16 ***
totwrk        -0.158079    0.020867 -7.5755 1.639e-13 ***
age            2.437082    1.671818  1.4577   0.14551    
south         78.046123   45.287154  1.7234   0.08542 .  
male          36.485386   43.350193  0.8416   0.40037    
smsa         -34.964643   38.231919 -0.9145   0.36086    
yngkid        50.136466   54.789835  0.9151   0.36058    
marr          54.071604   46.920123  1.1524   0.24968    
union         27.018684   43.044039  0.6277   0.53048    
log(hrwage)   -2.726776   32.202444 -0.0847   0.93255'''
date_string

'               Estimate  Std. Error t value  Pr(>|t|)    \n(Intercept) 3431.803558   96.309432 35.6331 < 2.2e-16 ***\ntotwrk        -0.158079    0.020867 -7.5755 1.639e-13 ***\nage            2.437082    1.671818  1.4577   0.14551    \nsouth         78.046123   45.287154  1.7234   0.08542 .  \nmale          36.485386   43.350193  0.8416   0.40037    \nsmsa         -34.964643   38.231919 -0.9145   0.36086    \nyngkid        50.136466   54.789835  0.9151   0.36058    \nmarr          54.071604   46.920123  1.1524   0.24968    \nunion         27.018684   43.044039  0.6277   0.53048    \nlog(hrwage)   -2.726776   32.202444 -0.0847   0.93255'

In [10]:
df = pd.read_csv(io.StringIO(date_string), sep='\s+')
df

Estimate       Std.    Error          t    value Pr(>|t|)
(Intercept)  3431.803558  96.309432  35.6331          <  2.2e-16      ***
totwrk         -0.158079   0.020867  -7.5755  1.639e-13      ***      NaN
age             2.437082   1.671818   1.4577    0.14551      NaN      NaN
south          78.046123  45.287154   1.7234    0.08542        .      NaN
male           36.485386  43.350193   0.8416    0.40037      NaN      NaN
smsa          -34.964643  38.231919  -0.9145    0.36086      NaN      NaN
yngkid         50.136466  54.789835   0.9151    0.36058      NaN      NaN
marr           54.071604  46.920123   1.1524    0.24968      NaN      NaN
union          27.018684  43.044039   0.6277    0.53048      NaN      NaN
log(hrwage)    -2.726776  32.202444  -0.0847    0.93255      NaN      NaN

In [11]:
df.rename(columns={'Std.':'std.errror','Error':'t_value'}, inplace=True)
df

Estimate  std.errror  t_value          t    value Pr(>|t|)
(Intercept)  3431.803558   96.309432  35.6331          <  2.2e-16      ***
totwrk         -0.158079    0.020867  -7.5755  1.639e-13      ***      NaN
age             2.437082    1.671818   1.4577    0.14551      NaN      NaN
south          78.046123   45.287154   1.7234    0.08542        .      NaN
male           36.485386   43.350193   0.8416    0.40037      NaN      NaN
smsa          -34.964643   38.231919  -0.9145    0.36086      NaN      NaN
yngkid         50.136466   54.789835   0.9151    0.36058      NaN      NaN
marr           54.071604   46.920123   1.1524    0.24968      NaN      NaN
union          27.018684   43.044039   0.6277    0.53048      NaN      NaN
log(hrwage)    -2.726776   32.202444  -0.0847    0.93255      NaN      NaN

In [12]:
df.drop(columns={'t','value','Pr(>|t|)'}, inplace=True)
df

Estimate  std.errror  t_value
(Intercept)  3431.803558   96.309432  35.6331
totwrk         -0.158079    0.020867  -7.5755
age             2.437082    1.671818   1.4577
south          78.046123   45.287154   1.7234
male           36.485386   43.350193   0.8416
smsa          -34.964643   38.231919  -0.9145
yngkid         50.136466   54.789835   0.9151
marr           54.071604   46.920123   1.1524
union          27.018684   43.044039   0.6277
log(hrwage)    -2.726776   32.202444  -0.0847

In [14]:
alpha= 0.1
t_critical = stats.t.ppf(1 - alpha/2,707-len(df)).round(3)
t_critical

1.647

### Ответ: 1.647

In [15]:
df['Znachimost'] = df.apply(lambda x: 'znachim' if np.abs(x['t_value'])>t_critical else 'ne znachim', axis=1 )
df

Estimate  std.errror  t_value  Znachimost
(Intercept)  3431.803558   96.309432  35.6331     znachim
totwrk         -0.158079    0.020867  -7.5755     znachim
age             2.437082    1.671818   1.4577  ne znachim
south          78.046123   45.287154   1.7234     znachim
male           36.485386   43.350193   0.8416  ne znachim
smsa          -34.964643   38.231919  -0.9145  ne znachim
yngkid         50.136466   54.789835   0.9151  ne znachim
marr           54.071604   46.920123   1.1524  ne znachim
union          27.018684   43.044039   0.6277  ne znachim
log(hrwage)    -2.726776   32.202444  -0.0847  ne znachim

## Задача 1.3 (wage equation 1)

In [16]:
data_string = '''Names         Estimate  Std.Error
(Intercept)    4.974      0.165
age            0.021      0.004
IQ             0.008      0.001
married        0.201      0.040
south         -0.099      0.027
urban          0.175      0.028'''
data_string

'Names         Estimate  Std.Error\n(Intercept)    4.974      0.165\nage            0.021      0.004\nIQ             0.008      0.001\nmarried        0.201      0.040\nsouth         -0.099      0.027\nurban          0.175      0.028'

In [17]:
df = pd.read_csv(io.StringIO(data_string), sep ='\s+')
df

Names  Estimate  Std.Error
0  (Intercept)     4.974      0.165
1          age     0.021      0.004
2           IQ     0.008      0.001
3      married     0.201      0.040
4        south    -0.099      0.027
5        urban     0.175      0.028

In [18]:
df['t_nabl'] = np.abs(df['Estimate']/df['Std.Error'])
df

Names  Estimate  Std.Error     t_nabl
0  (Intercept)     4.974      0.165  30.145455
1          age     0.021      0.004   5.250000
2           IQ     0.008      0.001   8.000000
3      married     0.201      0.040   5.025000
4        south    -0.099      0.027   3.666667
5        urban     0.175      0.028   6.250000

In [19]:
alpha = 0.05
t_crit = stats.t.ppf(1-0.05/2, 937 - (8+1)).round(3)
t_crit

1.963

### Ответ: 1.963

In [20]:
df['Znachimosty'] = df.apply(lambda x: 'znachim' if x['t_nabl']>t_crit else 'ne_znachim', axis = 1)
df

Names  Estimate  Std.Error     t_nabl Znachimosty
0  (Intercept)     4.974      0.165  30.145455     znachim
1          age     0.021      0.004   5.250000     znachim
2           IQ     0.008      0.001   8.000000     znachim
3      married     0.201      0.040   5.025000     znachim
4        south    -0.099      0.027   3.666667     znachim
5        urban     0.175      0.028   6.250000     znachim

## Задача 1.4 (wage equation 1, Python)

In [22]:
date_string = '''               Estimate  Std. Error t value  Pr(>|t|)    
(Intercept)  4.97400041  0.16544459 30.0644 < 2.2e-16 ***
age          0.02125292  0.00400766  5.3031 1.424e-07 ***
IQ           0.00820145  0.00084111  9.7508 < 2.2e-16 ***
married      0.20097997  0.04024862  4.9935 7.082e-07 ***
south       -0.09897822  0.02682522 -3.6897 0.0002375 ***
urban        0.17498055  0.02763721  6.3313 3.779e-10 ***'''
date_string

'               Estimate  Std. Error t value  Pr(>|t|)    \n(Intercept)  4.97400041  0.16544459 30.0644 < 2.2e-16 ***\nage          0.02125292  0.00400766  5.3031 1.424e-07 ***\nIQ           0.00820145  0.00084111  9.7508 < 2.2e-16 ***\nmarried      0.20097997  0.04024862  4.9935 7.082e-07 ***\nsouth       -0.09897822  0.02682522 -3.6897 0.0002375 ***\nurban        0.17498055  0.02763721  6.3313 3.779e-10 ***'

In [23]:
df = pd.read_csv(io.StringIO(date_string), sep='\s+')
df

Estimate      Std.    Error          t    value Pr(>|t|)
(Intercept)  4.974000  0.165445  30.0644          <  2.2e-16      ***
age          0.021253  0.004008   5.3031  1.424e-07      ***      NaN
IQ           0.008201  0.000841   9.7508          <  2.2e-16      ***
married      0.200980  0.040249   4.9935  7.082e-07      ***      NaN
south       -0.098978  0.026825  -3.6897  0.0002375      ***      NaN
urban        0.174981  0.027637   6.3313  3.779e-10      ***      NaN

In [24]:
df.rename(columns={'Std.':'std.errror','Error':'t_value'}, inplace=True)
df

Estimate  std.errror  t_value          t    value Pr(>|t|)
(Intercept)  4.974000    0.165445  30.0644          <  2.2e-16      ***
age          0.021253    0.004008   5.3031  1.424e-07      ***      NaN
IQ           0.008201    0.000841   9.7508          <  2.2e-16      ***
married      0.200980    0.040249   4.9935  7.082e-07      ***      NaN
south       -0.098978    0.026825  -3.6897  0.0002375      ***      NaN
urban        0.174981    0.027637   6.3313  3.779e-10      ***      NaN

In [25]:
df.drop(columns={'t','value','Pr(>|t|)'}, inplace=True)
df

Estimate  std.errror  t_value
(Intercept)  4.974000    0.165445  30.0644
age          0.021253    0.004008   5.3031
IQ           0.008201    0.000841   9.7508
married      0.200980    0.040249   4.9935
south       -0.098978    0.026825  -3.6897
urban        0.174981    0.027637   6.3313

In [26]:
alpha= 0.01
t_critical = stats.t.ppf(1 - alpha/2,935-len(df)).round(3)
t_critical

2.581

### Ответ: 2.581

In [28]:
df['Znachimost'] = df.apply(lambda x: 'znachim' if np.abs(x['t_value'])>t_critical else 'ne znachim', axis=1 )
df

Estimate  std.errror  t_value Znachimost
(Intercept)  4.974000    0.165445  30.0644    znachim
age          0.021253    0.004008   5.3031    znachim
IQ           0.008201    0.000841   9.7508    znachim
married      0.200980    0.040249   4.9935    znachim
south       -0.098978    0.026825  -3.6897    znachim
urban        0.174981    0.027637   6.3313    znachim

## Задача 1.5 (wage equation 2)

In [29]:
data_string = '''Names        Estimate   Std.Error
(Intercept)    1.250      0.061
exper          0.037      0.006
I(exper^2)    -0.001      0.000
female        -0.363      0.040
married        0.144      0.045
smsa           0.273      0.044'''
data_string

'Names        Estimate   Std.Error\n(Intercept)    1.250      0.061\nexper          0.037      0.006\nI(exper^2)    -0.001      0.000\nfemale        -0.363      0.040\nmarried        0.144      0.045\nsmsa           0.273      0.044'

In [30]:
df = pd.read_csv(io.StringIO(data_string), sep ='\s+')
df

Names  Estimate  Std.Error
0  (Intercept)     1.250      0.061
1        exper     0.037      0.006
2   I(exper^2)    -0.001      0.000
3       female    -0.363      0.040
4      married     0.144      0.045
5         smsa     0.273      0.044

In [31]:
df['t_nabl'] = np.abs(df['Estimate']/df['Std.Error'])
df

Names  Estimate  Std.Error     t_nabl
0  (Intercept)     1.250      0.061  20.491803
1        exper     0.037      0.006   6.166667
2   I(exper^2)    -0.001      0.000        inf
3       female    -0.363      0.040   9.075000
4      married     0.144      0.045   3.200000
5         smsa     0.273      0.044   6.204545

In [32]:
alpha = 0.05
t_crit = stats.t.ppf(1-0.05/2, 527 - (8+1)).round(3)
t_crit

1.965

### Ответ: 1.965

In [33]:
df['Znachimosty'] = df.apply(lambda x: 'znachim' if x['t_nabl']>t_crit else 'ne_znachim', axis = 1)
df

Names  Estimate  Std.Error     t_nabl Znachimosty
0  (Intercept)     1.250      0.061  20.491803     znachim
1        exper     0.037      0.006   6.166667     znachim
2   I(exper^2)    -0.001      0.000        inf     znachim
3       female    -0.363      0.040   9.075000     znachim
4      married     0.144      0.045   3.200000     znachim
5         smsa     0.273      0.044   6.204545     znachim

## Задача 1.6 (wage equation 2, Python)

In [34]:
date_string = '''               Estimate  Std. Error t value  Pr(>|t|)    
(Intercept)  1.25042289  0.06140967 20.3620 < 2.2e-16 ***
exper        0.03687966  0.00569646  6.4741 2.209e-10 ***
I(exper^2)  -0.00077167  0.00012195 -6.3280 5.366e-10 ***
female      -0.36273973  0.03974402 -9.1269 < 2.2e-16 ***
married      0.14378405  0.04549137  3.1607  0.001666 ** 
smsa         0.27256198  0.04411563  6.1784 1.308e-09 ***'''
date_string

'               Estimate  Std. Error t value  Pr(>|t|)    \n(Intercept)  1.25042289  0.06140967 20.3620 < 2.2e-16 ***\nexper        0.03687966  0.00569646  6.4741 2.209e-10 ***\nI(exper^2)  -0.00077167  0.00012195 -6.3280 5.366e-10 ***\nfemale      -0.36273973  0.03974402 -9.1269 < 2.2e-16 ***\nmarried      0.14378405  0.04549137  3.1607  0.001666 ** \nsmsa         0.27256198  0.04411563  6.1784 1.308e-09 ***'

In [35]:
df = pd.read_csv(io.StringIO(date_string), sep='\s+')
df

Estimate      Std.    Error          t    value Pr(>|t|)
(Intercept)  1.250423  0.061410  20.3620          <  2.2e-16      ***
exper        0.036880  0.005696   6.4741  2.209e-10      ***      NaN
I(exper^2)  -0.000772  0.000122  -6.3280  5.366e-10      ***      NaN
female      -0.362740  0.039744  -9.1269          <  2.2e-16      ***
married      0.143784  0.045491   3.1607   0.001666       **      NaN
smsa         0.272562  0.044116   6.1784  1.308e-09      ***      NaN

In [36]:
df.rename(columns={'Std.':'std.errror','Error':'t_value'}, inplace=True)
df

Estimate  std.errror  t_value          t    value Pr(>|t|)
(Intercept)  1.250423    0.061410  20.3620          <  2.2e-16      ***
exper        0.036880    0.005696   6.4741  2.209e-10      ***      NaN
I(exper^2)  -0.000772    0.000122  -6.3280  5.366e-10      ***      NaN
female      -0.362740    0.039744  -9.1269          <  2.2e-16      ***
married      0.143784    0.045491   3.1607   0.001666       **      NaN
smsa         0.272562    0.044116   6.1784  1.308e-09      ***      NaN

In [37]:
df.drop(columns={'t','value','Pr(>|t|)'}, inplace=True)
df

Estimate  std.errror  t_value
(Intercept)  1.250423    0.061410  20.3620
exper        0.036880    0.005696   6.4741
I(exper^2)  -0.000772    0.000122  -6.3280
female      -0.362740    0.039744  -9.1269
married      0.143784    0.045491   3.1607
smsa         0.272562    0.044116   6.1784

In [38]:
alpha= 0.05
t_critical = stats.t.ppf(1 - alpha/2, 527-len(df)).round(3)
t_critical

1.965

### Ответ: 1.965

In [39]:
df['Znachimost'] = df.apply(lambda x: 'znachim' if np.abs(x['t_value'])>t_critical else 'ne znachim', axis=1 )
df

Estimate  std.errror  t_value Znachimost
(Intercept)  1.250423    0.061410  20.3620    znachim
exper        0.036880    0.005696   6.4741    znachim
I(exper^2)  -0.000772    0.000122  -6.3280    znachim
female      -0.362740    0.039744  -9.1269    znachim
married      0.143784    0.045491   3.1607    znachim
smsa         0.272562    0.044116   6.1784    znachim

In [40]:
df['Znachimost'] = df.apply(lambda x: 'znachim' if np.abs(x['t_value'])>t_critical else 'ne znachim', axis=1 )
df

Estimate  std.errror  t_value Znachimost
(Intercept)  1.250423    0.061410  20.3620    znachim
exper        0.036880    0.005696   6.4741    znachim
I(exper^2)  -0.000772    0.000122  -6.3280    znachim
female      -0.362740    0.039744  -9.1269    znachim
married      0.143784    0.045491   3.1607    znachim
smsa         0.272562    0.044116   6.1784    znachim

## Задача 1.7 (cost equation 1)

In [41]:
data_string = '''Names        Estimate   Std.Error
(Intercept)   -7.472      1.063
log(q)         0.838      0.009
log(pl)        0.044      0.106
log(pk)        0.188      0.087
log(pf)        0.713      0.064'''
data_string

'Names        Estimate   Std.Error\n(Intercept)   -7.472      1.063\nlog(q)         0.838      0.009\nlog(pl)        0.044      0.106\nlog(pk)        0.188      0.087\nlog(pf)        0.713      0.064'

In [42]:
df = pd.read_csv(io.StringIO(data_string), sep ='\s+')
df

Names  Estimate  Std.Error
0  (Intercept)    -7.472      1.063
1       log(q)     0.838      0.009
2      log(pl)     0.044      0.106
3      log(pk)     0.188      0.087
4      log(pf)     0.713      0.064

In [43]:
df['t_nabl'] = np.abs(df['Estimate']/df['Std.Error'])
df

Names  Estimate  Std.Error     t_nabl
0  (Intercept)    -7.472      1.063   7.029163
1       log(q)     0.838      0.009  93.111111
2      log(pl)     0.044      0.106   0.415094
3      log(pk)     0.188      0.087   2.160920
4      log(pf)     0.713      0.064  11.140625

In [44]:
alpha = 0.05
t_crit = stats.t.ppf(1-0.05/2, 159 - 5).round(3)
t_crit

1.975

### Ответ: 1.975

In [46]:
df['Znachimosty'] = df.apply(lambda x: 'znachim' if x['t_nabl']>t_crit else 'ne_znachim', axis = 1)
df

Names  Estimate  Std.Error     t_nabl Znachimosty
0  (Intercept)    -7.472      1.063   7.029163     znachim
1       log(q)     0.838      0.009  93.111111     znachim
2      log(pl)     0.044      0.106   0.415094  ne_znachim
3      log(pk)     0.188      0.087   2.160920     znachim
4      log(pf)     0.713      0.064  11.140625     znachim

## Задача 1.8 (cost equation 2, Python)

In [48]:
date_string = '''               Estimate  Std. Error t value  Pr(>|t|)    
(Intercept) -7.472227   1.063238 -7.0278 6.491e-11 ***
log(q)       0.838111   0.009387 89.2845  2.2e-16 ***
log(pl)      0.044258   0.105820  0.4182   0.67636    
log(pk)      0.188449   0.087067  2.1644   0.03198 *  
log(pf)      0.713098   0.064355 11.0806  2.2e-16 ***'''
date_string

'               Estimate  Std. Error t value  Pr(>|t|)    \n(Intercept) -7.472227   1.063238 -7.0278 6.491e-11 ***\nlog(q)       0.838111   0.009387 89.2845  2.2e-16 ***\nlog(pl)      0.044258   0.105820  0.4182   0.67636    \nlog(pk)      0.188449   0.087067  2.1644   0.03198 *  \nlog(pf)      0.713098   0.064355 11.0806  2.2e-16 ***'

In [49]:
df = pd.read_csv(io.StringIO(date_string), sep='\s+')
df

Estimate      Std.     Error        t         value Pr(>|t|)
0  (Intercept) -7.472227  1.063238  -7.0278  6.491000e-11      ***
1       log(q)  0.838111  0.009387  89.2845  2.200000e-16      ***
2      log(pl)  0.044258  0.105820   0.4182  6.763600e-01      NaN
3      log(pk)  0.188449  0.087067   2.1644  3.198000e-02        *
4      log(pf)  0.713098  0.064355  11.0806  2.200000e-16      ***

In [50]:
df.rename(columns={'Std.':'std.errror','Error':'t_value'}, inplace=True)
df

Estimate  std.errror   t_value        t         value Pr(>|t|)
0  (Intercept)   -7.472227  1.063238  -7.0278  6.491000e-11      ***
1       log(q)    0.838111  0.009387  89.2845  2.200000e-16      ***
2      log(pl)    0.044258  0.105820   0.4182  6.763600e-01      NaN
3      log(pk)    0.188449  0.087067   2.1644  3.198000e-02        *
4      log(pf)    0.713098  0.064355  11.0806  2.200000e-16      ***

In [51]:
df.drop(columns={'t','value','Pr(>|t|)'}, inplace=True)
df

Estimate  std.errror   t_value
0  (Intercept)   -7.472227  1.063238
1       log(q)    0.838111  0.009387
2      log(pl)    0.044258  0.105820
3      log(pk)    0.188449  0.087067
4      log(pf)    0.713098  0.064355

In [52]:
alpha= 0.01
t_critical = stats.t.ppf(1 - alpha/2, 159-len(df)).round(3)
t_critical

2.608

### Ответ: 2.608

## 2.2 Задача (output equation 2, задача 2.1 в классной работе)

In [54]:
df_lab = pd.read_csv('https://raw.githubusercontent.com/artamonoff/Econometrica/master/python-notebooks/data-csv/Labour.csv')
df_lab

capital  labour     output       wage
0     2.606563     184   9.250759  43.080307
1     1.323237      91   3.664310  27.780016
2    22.093692     426  28.781516  44.467748
3    10.737851      72   4.124642  39.734710
4     1.161365      46   2.890150  34.650709
..         ...     ...        ...        ...
564   2.625403      20   1.424376  33.477545
565   1.276386      61   2.109048  26.300732
566   1.953869     117   6.241870  41.153979
567   1.318527      46   7.902237  66.720139
568   2.832391     161   6.804085  36.618435

[569 rows x 4 columns]

In [55]:
models1 = smf.ols(data = df_lab, formula = 'np.log(output)~np.log(capital)+np.log(labour)').fit()
models1.summary()

<class 'statsmodels.iolib.summary.Summary'>
"""
                            OLS Regression Results                            
==============================================================================
Dep. Variable:         np.log(output)   R-squared:                       0.838
Model:                            OLS   Adj. R-squared:                  0.837
Method:                 Least Squares   F-statistic:                     1462.
Date:                Tue, 28 Mar 2023   Prob (F-statistic):          2.64e-224
Time:                        23:12:05   Log-Likelihood:                -385.95
No. Observations:                 569   AIC:                             777.9
Df Residuals:                     566   BIC:                             790.9
Df Model:                           2                                         
Covariance Type:            nonrobust                                         
===================================================================================
                      coef    std err          t      P>|t|      [0.025      0.975]
-----------------------------------------------------------------------------------
Intercept          -1.7115      0.097    -17.697      0.000      -1.901      -1.522
np.log(capital)     0.2076      0.017     12.077      0.000       0.174       0.241
np.log(labour)      0.7148      0.023     30.890      0.000       0.669       0.760
==============================================================================
Omnibus:                      259.475   Durbin-Watson:                   1.963
Prob(Omnibus):                  0.000   Jarque-Bera (JB):             2677.293
Skew:                           1.734   Prob(JB):                         0.00
Kurtosis:                      13.045   Cond. No.                         24.3
==============================================================================

Notes:
[1] Standard Errors assume that the covariance matrix of the errors is correctly specified.
"""

In [56]:
models1.summary(alpha = 0.1).tables[1]

<class 'statsmodels.iolib.table.SimpleTable'>

In [57]:
models1.t_test('np.log(labour)=0.7')

<class 'statsmodels.stats.contrast.ContrastResults'>
                             Test for Constraints                             
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
c0             0.7148      0.023      0.642      0.521       0.669       0.760

In [58]:
beta_cap = 0.7148
ste_cap = 0.023
t_nabl = np.abs((beta_cap-0.7)/ste_cap).round(2)
t_nabl

0.64

$$
t_{nabl} =0.64
$$

In [59]:
t_crit = stats.t.ppf(1-0.1/2, len(df_lab)-3).round(2)
t_crit

1.65

$$
t_{crit}=1.65
$$

### ГИПОТЕЗА

$$
H_0:\beta_{labour}=0,5\\
H_1:\beta_{labour}\neq0,5
$$

### ВЫВОД:

## beta_{labour} значительно отличается от 0,7

## 2.3 Задача (cost equation 1)

In [63]:
df_lab = pd.read_csv('https://raw.githubusercontent.com/artamonoff/Econometrica/master/python-notebooks/data-csv/Electricity.csv')
df_lab

cost        q        pl      sl      pk      sk      pf      sf
0     0.2130      8.0   6869.47  0.3291  64.945  0.4197  18.000  0.2512
1     3.0427    869.0   8372.96  0.1030  68.227  0.2913  21.067  0.6057
2     9.4059   1412.0   7960.90  0.0891  40.692  0.1567  41.530  0.7542
3     0.7606     65.0   8971.89  0.2802  41.243  0.1282  28.539  0.5916
4     2.2587    295.0   8218.40  0.1772  71.940  0.1623  39.200  0.6606
..       ...      ...       ...     ...     ...     ...     ...     ...
153   6.8293    946.6  10642.16  0.0883  43.600  0.1914  51.463  0.7203
154   3.7605    377.0   7432.24  0.2117  74.120  0.2274  33.436  0.5609
155   3.9822    391.0   5826.04  0.1926  78.288  0.0924  44.633  0.7151
156  30.1880   5317.0   9586.63  0.0845  78.008  0.2009  41.840  0.7147
157  67.8562  15220.0   6986.09  0.1451  74.025  0.2596  25.662  0.5945

[158 rows x 8 columns]

In [64]:
models2 = smf.ols(data = df_lab, formula = 'np.log(cost)~np.log(q)+np.log(pl)+np.log(pk)+np.log(pf)').fit()
models2.summary()

<class 'statsmodels.iolib.summary.Summary'>
"""
                            OLS Regression Results                            
==============================================================================
Dep. Variable:           np.log(cost)   R-squared:                       0.982
Model:                            OLS   Adj. R-squared:                  0.982
Method:                 Least Squares   F-statistic:                     2107.
Date:                Tue, 28 Mar 2023   Prob (F-statistic):          1.25e-132
Time:                        23:20:00   Log-Likelihood:                 25.928
No. Observations:                 158   AIC:                            -41.86
Df Residuals:                     153   BIC:                            -26.54
Df Model:                           4                                         
Covariance Type:            nonrobust                                         
==============================================================================
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
Intercept     -7.4722      1.063     -7.028      0.000      -9.573      -5.372
np.log(q)      0.8381      0.009     89.284      0.000       0.820       0.857
np.log(pl)     0.0443      0.106      0.418      0.676      -0.165       0.253
np.log(pk)     0.1884      0.087      2.164      0.032       0.016       0.360
np.log(pf)     0.7131      0.064     11.081      0.000       0.586       0.840
==============================================================================
Omnibus:                       43.278   Durbin-Watson:                   1.458
Prob(Omnibus):                  0.000   Jarque-Bera (JB):              109.870
Skew:                           1.123   Prob(JB):                     1.39e-24
Kurtosis:                       6.413   Cond. No.                         866.
==============================================================================

Notes:
[1] Standard Errors assume that the covariance matrix of the errors is correctly specified.
"""

In [65]:
models2.summary(alpha = 0.05).tables[1]

<class 'statsmodels.iolib.table.SimpleTable'>

In [66]:
models2.t_test('np.log(q)=1')

<class 'statsmodels.stats.contrast.ContrastResults'>
                             Test for Constraints                             
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
c0             0.8381      0.009    -17.246      0.000       0.820       0.857

In [67]:
beta_cap = 0.8381
ste_cap = 0.009
t_nabl = np.abs((beta_cap-1)/ste_cap).round(2)
t_nabl

17.99

$$
t_{nabl} =17.99
$$

In [68]:
t_crit = stats.t.ppf(1-0.05/2, len(df_lab)-3).round(2)
t_crit

1.98

$$
t_{crit}=1.98
$$

### ГИПОТЕЗА

$$
H_0:\beta_{q}=1\\
H_1:\beta_{q}\neq1
$$

### ВЫВОД:

## beta_{q} значительно отличается от 1